R/rvar-.R

```diff
@@ -546,10 +546,17 @@ copy_dimnames <- function(src, src_i, dst, dst_i) {
 # helpers: applying functions over rvars ----------------------------------
 
 # apply a summary function within each draw of the rvar (dropping other dimensions)
-summarise_rvar_within_draws <- function(x, .f, ...) {
+summarise_rvar_within_draws <- function(x, .f, ..., .transpose = FALSE, .when_empty = .f()) {
```

Comment:
is it safe to call all kinds of `.f` without input? Perhaps set `.when_empty` to a safer default?

Reply:
the intent here is that the `rvar_` versions of functions should yield similar results (or similar errors) to base functions given empty input. So e.g.:

```
> sum()
[1] 0
> rvar_sum()
rvar<1>[1] mean ± sd:
[1] 0 ± NA 
> mean()
Error in mean.default() : argument "x" is missing, with no default
> rvar_mean()
Error in mean.default() : argument "x" is missing, with no default 
```

That said, in some cases the default error message is a little obscure; for example:

```
> sd()
 Error in is.data.frame(x) : argument "x" is missing, with no default 
> rvar_sd()
 Error in is.data.frame(x) : argument "x" is missing, with no default 
```

Throwing an error is still the correct behavior, but I think we can probably provide a better error message via `.when_empty`

Reply:
I've updated it so that the functions that should throw errors now have more sensible error messages. Specifically `rvar_mean()`, `rvar_median()`, `rvar_sd()`, `rvar_var()`, and `rvar_mad()` now follow this pattern:

```
> rvar_median()
 Error: in rvar_median(): cannot take median of empty vector 
```

Does that sound good?

Reply:
Sounds good to me.

Reply:
Nice, these error messages are much better than the ones in base R